### DISCUSSION TITLE
Custom Bom/Build Order/Sales order name

### DISCUSSION THREAD

**Discussion by method1020** [inventree/InvenTree#4221] [Q&A] [answered]
Is it possible to make the so#/bo#/bom# name a user entry field instead of automatically generated?

**Comment by miggland** [comment, marked as answer]
It is already.. However, the reference has to match a pattern. The pattern is given in the settings, see Build/Purchase/Sales order page, and change "Order Reference Pattern".

When creating an order, you can change the reference in the given field.

**Reply by miggland** [reply]
See also https://docs.inventree.org/en/latest/build/build/#build-order-reference
